mobile_verifier/src/rewarder.rs

```diff
@@ -532,115 +499,28 @@ pub async fn reward_dc(
-
-    // write out any unallocated mapping rewards
-    let unallocated_mapping_reward_amount = total_mappers_pool
-        .round_dp_with_strategy(0, RoundingStrategy::ToZero)
-        .to_u64()
-        .unwrap_or(0)
-        - allocated_mapping_rewards;
-    write_unallocated_reward(
-        &mobile_rewards,
-        UnallocatedRewardType::Mapper,
-        unallocated_mapping_reward_amount,
-        reward_info,
-    )
-    .await?;
-
-    Ok(())
-}
-
-pub async fn reward_oracles(
-    mobile_rewards: FileSinkClient<proto::MobileRewardShare>,
-    reward_info: &EpochRewardInfo,
-) -> anyhow::Result<()> {
-    // atm 100% of oracle rewards are assigned to 'unallocated'
-    let total_oracle_rewards =
-        reward_shares::get_scheduled_tokens_for_oracles(reward_info.epoch_emissions);
-    let allocated_oracle_rewards = 0_u64;
-    let unallocated_oracle_reward_amount = total_oracle_rewards
-        .round_dp_with_strategy(0, RoundingStrategy::ToZero)
-        .to_u64()
-        .unwrap_or(0)
-        - allocated_oracle_rewards;
-    write_unallocated_reward(
-        &mobile_rewards,
-        UnallocatedRewardType::Oracle,
-        unallocated_oracle_reward_amount,
-        reward_info,
-    )
-    .await?;
-    Ok(())
-}
-
 pub async fn reward_service_providers(
-    dc_sessions: ServiceProviderDCSessions,
-    sp_promotions: ServiceProviderPromotions,
     mobile_rewards: FileSinkClient<proto::MobileRewardShare>,
     reward_info: &EpochRewardInfo,
-    hnt_bone_price: Decimal,
 ) -> anyhow::Result<()> {
-    use service_provider::ServiceProviderRewardInfos;
-
     let total_sp_rewards = service_provider::get_scheduled_tokens(reward_info.epoch_emissions);
-
-    let sps = ServiceProviderRewardInfos::new(
-        dc_sessions,
-        sp_promotions,
-        total_sp_rewards,
-        hnt_bone_price,
-        reward_info.clone(),
-    );
-
-    let mut unallocated_sp_rewards = total_sp_rewards
+    let sp_reward_amount = total_sp_rewards
         .round_dp_with_strategy(0, RoundingStrategy::ToZero)
         .to_u64()
         .unwrap_or(0);
 
-    for (amount, reward) in sps.iter_rewards() {
-        unallocated_sp_rewards -= amount;
-        mobile_rewards.write(reward, []).await?.await??;
-    }
-
-    // write out any unallocated service provider reward
-    write_unallocated_reward(
-        &mobile_rewards,
-        UnallocatedRewardType::ServiceProvider,
-        unallocated_sp_rewards,
-        reward_info,
-    )
-    .await?;
+    // Write a single ServiceProviderReward for HeliumMobile with the full 24% allocation
+    let sp_reward = proto::MobileRewardShare {
+        start_period: reward_info.epoch_period.start.encode_timestamp(),
+        end_period: reward_info.epoch_period.end.encode_timestamp(),
+        reward: Some(ProtoReward::ServiceProviderReward(
+            proto::ServiceProviderReward {
+                service_provider_id: ServiceProvider::HeliumMobile as i32,
```

Comment:
using `into()` feels like more preferable way to convert enum to int.
But result still should be the same 
```suggestion
                service_provider_id: ServiceProvider::HeliumMobile.into(),
```